pkg/services/ngalert/migration/channel.go

```diff
@@ -180,7 +94,7 @@ func (om *OrgMigration) createNotifier(c *legacymodels.AlertNotification) (*apim
 	}
 
 	return &apimodels.PostableGrafanaReceiver{
-		UID:                   uid,
+		UID:                   c.UID,
```

Comment:
I am not sure I understand what this change does. Previously, we assumed that UID of legacy notifier could be empty and generated one in that case. This change seems to discard this logic. Was the assumption incorrect? Do we want to tolerate empty UIDs?

Reply:
Explained below: https://github.com/grafana/grafana/pull/76527#discussion_r1364324654